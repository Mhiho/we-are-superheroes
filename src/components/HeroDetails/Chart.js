import React from "react";
import "../../index.css";

const Chart = (props) => {
  return (
    <div className="chart-container d-flex justify-content-center">
      <div className="chart animation">
        <p className="text-center">{props.percentage}</p>
        <div
          style={{
            background: "#363636",
            width: "30px",
            margin: "auto",
            height: `${100 - props.percentage}px`,
          }}
          className="resize"
        ></div>
        <div
          style={{
            background: "gold",
            width: "30px",
            height: `${props.percentage}px`,
            margin: "auto",
          }}
          className="chart-column"
        ></div>
        <h6 className="text-center">{props.label}</h6>
      </div>
    </div>
  );
};
export default Chart;
