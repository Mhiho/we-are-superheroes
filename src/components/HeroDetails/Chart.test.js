import React from 'react';
import renderer from 'react-test-renderer';

import Chart from './Chart';

it('renders correctly when there is no props', () => {
  const tree = renderer.create(<Chart />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly when there is percentage', () => {
    const percentage = 'percentage';
    const tree = renderer.create(<Chart percentage={percentage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly when there is label', () => {
    const label = 'label';
    const tree = renderer.create(<Chart label={label} />).toJSON();
    expect(tree).toMatchSnapshot();
  });