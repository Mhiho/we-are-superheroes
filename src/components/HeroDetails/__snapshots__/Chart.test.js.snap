// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is label 1`] = `
<div
  className="chart-container d-flex justify-content-center"
>
  <div
    className="chart animation"
  >
    <p
      className="text-center"
    />
    <div
      className="resize"
      style={
        Object {
          "background": "#363636",
          "height": "NaNpx",
          "margin": "auto",
          "width": "30px",
        }
      }
    />
    <div
      className="chart-column"
      style={
        Object {
          "background": "gold",
          "height": "undefinedpx",
          "margin": "auto",
          "width": "30px",
        }
      }
    />
    <h6
      className="text-center"
    >
      label
    </h6>
  </div>
</div>
`;

exports[`renders correctly when there is no props 1`] = `
<div
  className="chart-container d-flex justify-content-center"
>
  <div
    className="chart animation"
  >
    <p
      className="text-center"
    />
    <div
      className="resize"
      style={
        Object {
          "background": "#363636",
          "height": "NaNpx",
          "margin": "auto",
          "width": "30px",
        }
      }
    />
    <div
      className="chart-column"
      style={
        Object {
          "background": "gold",
          "height": "undefinedpx",
          "margin": "auto",
          "width": "30px",
        }
      }
    />
    <h6
      className="text-center"
    />
  </div>
</div>
`;

exports[`renders correctly when there is percentage 1`] = `
<div
  className="chart-container d-flex justify-content-center"
>
  <div
    className="chart animation"
  >
    <p
      className="text-center"
    >
      percentage
    </p>
    <div
      className="resize"
      style={
        Object {
          "background": "#363636",
          "height": "NaNpx",
          "margin": "auto",
          "width": "30px",
        }
      }
    />
    <div
      className="chart-column"
      style={
        Object {
          "background": "gold",
          "height": "percentagepx",
          "margin": "auto",
          "width": "30px",
        }
      }
    />
    <h6
      className="text-center"
    />
  </div>
</div>
`;
